### ISSUE TITLE
Improve screen reading of character count variation of text input

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/vets-design-system-documentation#2350]
## Description

When testing the [**Character count**](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2323#issuecomment-1862849659) with Mac/VoiceOver, we discovered that our version does not read out the text being input, current value, and the remaining characters left in the same way. See the video.
- Our [**Character Count**](https://design.va.gov/storybook/?path=/docs/uswds-va-text-input--default#with-character-count) variation does work in the same way as [USWDS](https://designsystem.digital.gov/components/character-count/), however, it seems to cut-off when reading (at least in VoiceOver, USWDS seemed to perform better, I'll post a video). This may be something to do with the `aria-live` attributes that we should troubleshoot.
  - In the video you can see when I enter "123" on USWDS, it says, "1", "12", "24 characters left. With the VA version, it speaks out of order and interrupts by saying "9 characters lef", "1", "2", "8 characters lef", "12".
  https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/6180f260-0aa2-44f7-9fe8-4fb462b60e47
  

This will require some troubleshooting to see what differs in ours vs. theirs.

## Tasks
- [ ] Troubleshoot the character count variation
- [ ] Identify and fix any differences between ours and USWDS

## Acceptance Criteria
- [ ] Mac/VoiceOver reads the information out similarly to USWDS without cutting off information

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2350?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc
